src/main/java/org/skriptlang/skript/registry/SyntaxRegistry.java

```diff
@@ -0,0 +1,37 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.registry;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.Unmodifiable;
+
+import java.util.Set;
+
+interface SyntaxRegistry<I extends SyntaxInfo<?>> {
```

Comment:
IMO we should call this `SyntaxRegister` - as each register contains a specific record (in Skript's case a type of SyntaxInfo) and the registry itself (SkriptRegistry) contains all of the specific registers

Reply:
Okay I will change that.